db_client/alembic/versions/0066_add_import_id_to_slug_table_so_.py

```
@@ -0,0 +1,52 @@
+"""Add import id to slug table, so generated slugs can also be associated with collections
+
+Revision ID: 0066
+Revises: 0065
+Create Date: 2025-04-24 16:11:15.414259
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "0066"
+down_revision = "0065"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column("slug", sa.Column("collection_import_id", sa.Text(), nullable=True))
+    op.create_foreign_key(
+        op.f("fk_slug__collection_import_id__collection"),
+        "slug",
+        "collection",
+        ["collection_import_id"],
+        ["import_id"],
+    )
+    # Drop the existing CheckConstraint if it exists
+    op.drop_constraint("must_reference_exactly_one_entity", "slug", type_="check")
```

clarification: does this fail silently if the constraint it missing?

---

quite the opposite, if you try and insert a slug without one of the fks (detailed below) sql will complain.